backend/core/detection.py

```diff
@@ -526,31 +591,15 @@ def commit_hijack(self, monitor_event: Dict,
                 'timestamp_of_config': self.timestamp
             }
 
+            # identify the number of infected ases
             hijack_value['asns_inf'] = set()
-            # for squatting, all ASes except the origin are considered infected
-            if hij_type == 'Q':
-                if monitor_event['path']:
-                    hijack_value['asns_inf'] = set(monitor_event['path'][:-1])
-            # for sub-prefix hijacks, the infection depends on whether the
-            # hijacker is the origin/neighbor/sth else
-            elif hij_type == 'S':
-                if len(monitor_event['path']) > 1:
-                    if hijacker == monitor_event['path'][-1]:
-                        hijack_value['asns_inf'] = set(
-                            monitor_event['path'][:-1])
-                    elif hijacker == monitor_event['path'][-2]:
-                        hijack_value['asns_inf'] = set(
-                            monitor_event['path'][:-2])
-                    else:
-                        # assume the hijacker does a Type-2
-                        if len(monitor_event['path']) > 2:
-                            hijack_value['asns_inf'] = set(
-                                monitor_event['path'][:-3])
-            # for exact-prefix type-0/type-1 hijacks, the pollution depends on
-            # the type
-            else:
+            if hij_dimensions[1] in ['0', '1']:
```

Comment:
make this a set for O(1) lookup {'0', '1'}. No that it will have a difference with a list of two elements but it's better in practice!

Reply:
Fixed in updated version